report_thesis/src/sections/background.tex

```diff
@@ -316,7 +316,59 @@ \subsubsection{Extra Trees Regressor (ETR)}
+
+However, in order to explain \gls{gbr}, it is helpful to build on the concepts of ensemble learning and decision trees.
+Ensemble learning is a technique in machine learning where multiple models, known as \textit{weak learners}, are combined to produce more accurate predictions.
+Mathematically, ensemble learning can be defined as combining the predictions of $M$ weak learners to form a final prediction $\hat{y}$, such that:
+\begin{equation}
+    \hat{y} = \sum_{m=1}^{M} \alpha_m \hat{y}_m,
+\end{equation}
+where $\hat{y}_m$ is the prediction of the $m$-th weak learner and $\alpha_m$ is the weight assigned to the $m$-th weak learner.
+While there are various choices for weak learners, decision trees are a common choice.
+Decision trees are a core component of gradient boosting methods. 
+They partition the data into subsets based on feature values, aiming to create groups where data points have similar predicted outcomes. 
+The quality of a split is measured using various metrics. Some common metrics include impurity, information gain, and Gini index. 
+
+Once optimal splits are identified, the tree is constructed by repeatedly partitioning the data until a stopping criterion is met.
+The final model consists of splits that create distinct regions, each with a predicted response value based on the mean of the observations in that region.
+
+This leads us to \textit{Gradient Boosting}.
+Instead of building a single decision tree, gradient boosting constructs multiple trees sequentially, with each new tree correcting the errors of the previous ones.
+Each tree is small, with few terminal nodes, preventing large adjustments based on a single tree's predictions. 
+It also ensures that each tree makes small and simple error corrections, such that each step refines the model's performance more reliably.
+Finally, small trees have the property of high bias, meaning that they are unable to capture complex relationships in the data and low variance.
+By combining multiple small trees, the model can capture complex relationships while maintaining a low variance.
+
+The model is constructed iteratively by adding trees to correct the residuals of the previous trees. 
+Initially, the prediction is set as $\hat{f}_0(x) = 0$ and residuals as $r_i = y_i$ for all $i$ in the training set, where $x$ represents the input feature(s) and $y$ is the true value or target variable. 
+At each iteration $b$, predictions are updated as:
+$$
+    \hat{f}(x) = \hat{f}(x) + \lambda \hat{f}_b(x)
+$$
+where $\hat{f}_b(x)$ is the prediction of the $b$-th tree and $\lambda$ is the learning rate. Residuals are updated as:
+$$
+    r_i = y_i - \hat{f}(x)
+$$
+Each tree is then trained on these updated residuals. 
+This process of fitting a weak learner to predict the residuals and using its predictions to update the model is repeated for $B$ iterations, culminating in the final model:
```

Comment:
Where does $B$ come from? Is it a hyperparameter? So, we create $B$ trees total?

Reply:
Addressed.